inc/ruleticket.class.php

```diff
@@ -255,7 +257,17 @@ function executeActions($output, $params, array $input = []) {
                      $value[$actions[$action->fields["field"]]["appendtoarrayfield"]]
                             = $action->fields["value"];
                   }
-                  $output[$actions[$action->fields["field"]]["appendto"]][] = $value;
+
+                  // special case of appliance
+                  if ($action->fields["field"] === "assign_appliance") {
+                     if (!array_key_exists("items_id", $output)
+                        || (array_key_exists("items_id", $output) && !is_array($output['items_id']))) {
+                        $output["items_id"] = [];
+                     }
```

Comment:
If `$output['items_id']` has an invalid type, it is not a normal situation. Calling `trigger_error` will add a trace in GLPI log, which could help to fix this problem.
```suggestion
                     if (!array_key_exists("items_id", $output) || !is_array($output['items_id'])) {
                        if (!is_array($output['items_id'])) {
                           trigger_error('Invalid "items_id" output variable found in "assign_appliance" rule.', E_USER_WARNING);
                        }
                        $output["items_id"] = [];
                     }
```

Reply:
done